packages/fast-components/src/button/button.ts

```diff
@@ -0,0 +1,158 @@
+import { attr, FastElement, html, ref } from "@microsoft/fast-element";
+import { FormAssociated } from "../form-associated";
+
+export type ButtonAppearance =
+    | "accent"
+    | "lightweight"
+    | "neutral"
+    | "outline"
+    | "stealth";
+
+export const buttonTemplate = html<Button>`
+    <button
+        class="control"
+        ?autofocus=${x => x.autofocus}
+        ?disabled=${x => x.disabled}
+        form=${x => x.formId}
+        formaction=${x => x.formaction}
+        formenctype=${x => x.formenctype}
+        formmethod=${x => x.formmethod}
+        formnovalidate=${x => x.formnovalidate}
+        formtarget=${x => x.formtarget}
+        name=${x => x.name}
+        type=${x => x.type}
+        value=${x => x.value}
+    >
+        <span
+            part="start"
+            ${ref("startContainer")}
+        >
+            <slot
+                name="start"
+                ${ref("start")}
+                @slotchange=${x => x.handleStartContentChange()}
+            ></slot>
+        </span>
+        <span class="content" part="content">
+            <slot></slot>
+        </span>
+        <span
+            part="end"
+            ${ref("endContainer")}
+        >
+            <slot
+                name="end"
+                ${ref("end")}
+                @slotchange=${x => x.handleEndContentChange()}
+            ></slot>
+        </span>
+    </button>
+`;
+
+export class Button extends FormAssociated<HTMLInputElement> {
+    @attr
+    public appearance: ButtonAppearance = "neutral";
+    public appearanceChanged(
+        oldValue: ButtonAppearance,
+        newValue: ButtonAppearance
+    ): void {
+        if (oldValue !== newValue) {
+            this.classList.add(`${newValue}`);
+            this.classList.remove(`${oldValue}`);
+        }
+    }
+
+    @attr({ mode: "boolean" })
+    public autofocus: boolean;
+
+    @attr({ attribute: "form" })
+    public formId: string;
+
+    @attr
+    public formaction: string;
```

Comment:
Can these be camel case? This is hard to read.

Reply:
Actually, these all seem to shadow native properties. Is there any way we can promote those or declare them in one place, similarly to what I think we had from React's native HTML elements. I can see benefits to proxying the underlying attributes, but it also seems potentially limiting if new features are added or if we don't add an uncommon one and someone wants to use it. Do they have to create a custom override component?

Reply:
Yes, these shadow native properties. We are not at this time able to extend from the interface of an HTMLButtonElement because it will not work across all browsers. It's possible that we'll have that support in the future, but currently we need to declare and shadow the native API.